server/src/instant/storage/s3.clj

```diff
@@ -90,11 +93,32 @@
 ;; Instant <> S3 integration
 ;; ----------------------
 
-(defn upload-file-to-s3 [{:keys [app-id location-id] :as ctx} file]
+(defn upload-file-to-s3 [{:keys [app-id location-id content-type] :as ctx} file]
   (when (not (instance? java.io.InputStream file))
     (throw (Exception. "Unsupported file format")))
-  (let [ctx* (assoc ctx :object-key (->object-key app-id location-id))]
-    (s3-util/upload-stream-to-s3 (s3-async-client) bucket-name ctx* file)))
+  (if content-type
+    ;; Content-type provided
+    (let [ctx* (assoc ctx
+                      :object-key (->object-key app-id location-id)
+                      :content-type content-type)]
+      (s3-util/upload-stream-to-s3 (s3-async-client) bucket-name ctx* file))
+    ;; No content-type
+    (if (contains? (flags/flag :tika-enabled-apps) app-id)
+      (let [tika-stream (TikaInputStream/get file)
+            tika (Tika.)
```

Comment:
Do we need a new `Tika` for every invocation? I think it's safe to do `(def ^Tika detector (Tika.))` higher up.